lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb

```diff
@@ -243,28 +243,30 @@ def add_files_to_build_phases(native_target, test_native_targets)
           #
           def add_test_app_host_targets(test_native_targets)
             target.test_specs.map do |test_spec|
```

Comment:
This will be a bigger change. I had thought the same. Since there is only one app host per test type this should move to `pods_project_generator` instead.